### PR TITLE
legacy api refactorings

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | #3208 
| Refs tickets      | #3214 
| License           | MIT
| Changelog updated | no

### CODE REVIEW

File: src/system/AdminModule/Api/AdminApi.php [hunk 1]
```diff
@@ -41,15 +42,17 @@ public function create($args)
         // Argument check
         if (!isset($args['name']) || !strlen($args['name']) ||
             !isset($args['description'])) {
-            throw new \InvalidArgumentException(__('Invalid arguments array received'));
+            $translator = ServiceUtil::get('translator.default');
```
Comment (craigh) [inline src/system/AdminModule/Api/AdminApi.php]: ServiceUtil is deprecated.

Comment (Guite) [inline src/system/AdminModule/Api/AdminApi.php]: AdminApi is deprecated, too.

File: src/docs/Core-2.0/Categories/CategoriesHelper.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+Contains methods used for performing or amending the sorting of categories.
+
+The class makes the following methods available:
+
+    - cmpName($catA, $catB)
+    - cmpDesc($catA, $catB)
+    - sortByLocale(&$cats, $func = 'cmpName')
+    - resequence($cats, $step = 1)
+    - _tree_insert(&$tree, $entry, $currentpath = null)
```
Comment (craigh) [inline src/docs/Core-2.0/Categories/CategoriesHelper.md]: I'm not a fan of method names preceded by an underscore. this is a relic from old when PHP did not have private methods. Either make it private (and remove from this list) or make it public with a proper name.

Comment (craigh) [inline src/docs/Core-2.0/Categories/CategoriesHelper.md]: also, rename methods in new Apis to be descriptive - e.g. cmpName() to compareName() etc.

**Pull request merged**